src/components/TextInputs/PinCodeTextInput.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { Component } from 'react'
+import { StyleSheet, TextInputProps, View } from 'react-native'
+import NumberInput from './NumberInput'
+
+interface Props extends TextInputProps {
+  onChangeInput: (input: string) => void
+  authCode?: string
+}
+
+interface State {
+  digits: string[]
+  focused: boolean
+}
+
+export default class PinCodeTextInput extends Component<Props, State> {
+  private inputRefs: NumberInput[]
+
+  constructor(props: Props, state: State) {
+    super(props, state)
+    this.inputRefs = []
+    this.state = {
+      digits: ['', '', '', '', '', ''],
+      focused: false,
+    }
+  }
+
+  public static getDerivedStateFromProps(nextProps: Props, prevState: State) {
+    if (nextProps.authCode === prevState.digits.join('') || nextProps.authCode === undefined) {
+      return prevState
+    }
+
+    return {
+      digits: nextProps.authCode!.split(''),
+    }
+  }
+
+  public render() {
+    return (
+      <View style={styles.pinNumbersContainer}>
+        {[0, 1, 2, 3, 4, 5].map(i => (
+          <NumberInput
+            {...this.props}
+            key={i}
+            ref={component => (this.inputRefs[i] = component!)}
+            onChangeText={text => {
+              this.onChangeDigit(i, text)
+            }}
+            digit={this.state.digits[i]}
+          />
+        ))}
+      </View>
+    )
+  }
+
+  public onChangeDigit = (index: number, text: string) => {
+    this.state.digits[index] = text
+    this.props.onChangeInput(this.state.digits.join(''))
+    this.updatePinCodeInputFocus(index, this.inputRefs, text)
+  }
+
+  public updatePinCodeInputFocus = (index: number, inputRefs: NumberInput[], text: string) => {
+    if (text.length === 0 && index !== 0) {
```

Comment:
```suggestion
    if (text.length === 0 && index > 0) {
```

not a big deal but semantically `> 0` would make more sense